config/tls_config_test.go

```diff
@@ -58,7 +58,7 @@ func TestValidTLSConfig(t *testing.T) {
 		}
 		// non-nil functions are never equal.
 		got.GetClientCertificate = nil
-		if !reflect.DeepEqual(*got, *cfg.config) {
+		if !reflect.DeepEqual(got, cfg.config) {
```

Comment:
We shouldn't compare actual structs which contain mutexes. Compare using the pointers instead.